backend/src/user.rs

```diff
@@ -0,0 +1,39 @@
+use crate::database::{models::User, Database};
+use rocket::{
+    get,
+    serde::{json::Json, Serialize},
+};
+
+#[derive(Serialize)]
+pub enum UserError {
+    UserNotFound,
+}
+
+#[derive(Serialize)]
+pub struct ReturnUser {
+    email: String,
+    zid: String,
+    display_name: String,
+    degree_name: String,
+    degree_starting_year: i32,
+}
+
+#[get("/<user_id>")]
+pub async fn get_user(
```

Comment:
It might be worth being a little careful with this route -- it is certainly vulnerable to IDOR attacks, and tying a zID with a personal email might not be okay to some people, although the rest of the details should already be `acc`-able.

Not saying we need to scrap this route or remove the field, but it's important to consider the implications imo